net/ser2net/files/ser2net.init

```diff
@@ -184,3 +184,12 @@ start_service() {
 	procd_set_param command "$PROG" -n -c "$DYNAMICCFGFILE"
 	procd_close_instance
 }
+
+service_triggers() {
+        procd_add_reload_trigger "ser2net"
+}
+
+reload_service() {
+        rc_procd stop_service "$@"
+        rc_procd start_service "$@"
```

Comment:
why have you added this? if you don't implement reload, you get a stop/start implicitly.  You only need to implement reload if are doing something special.

Reply:
without that part the daemon did not restart. (I checked, the pid was always the same)

Reply:
you are probably missing a procd_set_param file /etc/config/ser2net ?

Reply:
https://openwrt.org/docs/guide-developer/procd-init-scripts
You need to add `procd_set_param file "$DYNAMICCFGFILE"` before `procd_set_param command` so that procd knows it should restart after config file changed.

Reply:
so basically keep service_triggers() and add the line you mentioned, am I correct?

Reply:
Yes